### PR TITLE
Add config-file option to get cluster

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Implement `--config-file` for `get cluster(s)` https://github.com/weaveworks/eksctl/issues/1126

### Tests

```
eksctl get clusters
2021-11-17 13:21:35 [ℹ]  eksctl version 0.75.0-dev+026bc14e.2021-11-17T13:20:12Z
2021-11-17 13:21:35 [ℹ]  using region us-west-2
NAME                            REGION          EKSCTL CREATED
cp-test8                        us-west-2       False
nm-ipv6-16nov-4                 us-west-2       True
test-nodegroup-cluster-config   us-west-2       True
➜  simple-cluster eksctl get clusters -f cluster-managed.yaml
2021-11-17 13:21:53 [ℹ]  eksctl version 0.75.0-dev+026bc14e.2021-11-17T13:20:12Z
2021-11-17 13:21:53 [ℹ]  using region us-west-2
NAME                            VERSION STATUS  CREATED                 VPC                     SUBNETS                              SECURITYGROUPS           PROVIDER
test-nodegroup-cluster-config   1.20    ACTIVE  2021-11-16T11:47:56Z    vpc-02ddfb0b9219aec75   subnet-052f10136984099c3,subnet-07ee321786eabaa1e,subnet-0a75be6d95e82d784,subnet-0abf9eefe70a7d22d,subnet-0dbfcd55c12c69311,subnet-0f320acfb4a56d4c8 sg-0daeb9b89b23eefac EKS
➜  simple-cluster eksctl get cluster --help
Get cluster(s)

Usage: eksctl get cluster [flags]

Aliases: cluster, clusters

General flags:
  -n, --name string          EKS cluster name
  -A, --all-regions          List clusters across all supported regions
  -r, --region string        AWS region
      --chunk-size int       return large lists in chunks rather than all at once, pass 0 to disable (default 100)
  -o, --output string        specifies the output format (valid option: table, json, yaml) (default "table")
      --timeout duration     maximum waiting time for any long-running operation (default 25m0s)
  -f, --config-file string   load configuration from a file (or stdin if set to '-')

AWS client flags:
  -p, --profile string   AWS credentials profile to use (overrides the AWS_PROFILE environment variable)

Common flags:
  -C, --color string   toggle colorized logs (valid options: true, false, fabulous) (default "true")
  -h, --help           help for this command
  -v, --verbose int    set log level, use 0 to silence, 4 for debugging and 5 for debugging with AWS debug logging (default 3)

Use 'eksctl get cluster [command] --help' for more information about a command.

➜  simple-cluster eksctl get cluster -A
NAME                            REGION          EKSCTL CREATED
bottlerocket-test               ca-central-1    True
cp-test                         ap-northeast-2  True
cp-test101                      ap-northeast-2  True
cp-test8                        us-west-2       False
nm-ipv6-16nov-4                 us-west-2       True
nmcfg2                          eu-west-2       True
pcluster                        me-south-1      True
test-nodegroup-cluster-config   us-west-2       True
windows-ipam                    ca-central-1    True
➜  simple-cluster eksctl get cluster -r eu-west-2
2021-11-17 13:23:18 [ℹ]  eksctl version 0.75.0-dev+026bc14e.2021-11-17T13:20:12Z
2021-11-17 13:23:18 [ℹ]  using region eu-west-2
NAME    REGION          EKSCTL CREATED
nmcfg2  eu-west-2       True
➜  simple-cluster eksctl get cluster -n test-nodegroup-cluster-config -f cluster-managed.yaml
Error: cannot use --name when --config-file/-f is set
```

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/ctl/cmdutils/configfile.go [hunk 1]
```diff
@@ -883,6 +883,14 @@ func NewGetLabelsLoader(cmd *Cmd, ngName string) ClusterConfigLoader {
 	return l
 }
 
+// NewGetClusterLoader will load config or use flags for 'eksctl get cluster(s)'
+func NewGetClusterLoader(cmd *Cmd) ClusterConfigLoader {
+	l := newCommonClusterConfigLoader(cmd)
+	l.flagsIncompatibleWithoutConfigFile.Delete("name")
```
Comment (cPu1) [inline pkg/ctl/cmdutils/configfile.go:889]: This has no effect as `flagsIncompatibleWithoutConfigFile` does not contain a `name` element.  
```suggestion

```

File: pkg/ctl/cmdutils/configfile.go [hunk 2]
```diff
@@ -883,6 +883,14 @@ func NewGetLabelsLoader(cmd *Cmd, ngName string) ClusterConfigLoader {
...
+
+  return l
```
[the remark this answers is not in the document]
Comment (Skarlso) [reply pkg/ctl/cmdutils/configfile.go]: yeah, merge error :D

File: pkg/ctl/cmdutils/configfile.go (hunk 1 above)
Comment (Skarlso) [reply pkg/ctl/cmdutils/configfile.go:889]: Huh, it doesn't? I think I had to add this otherwise it wasn't working... I'll test again.

Comment (cPu1) [reply pkg/ctl/cmdutils/configfile.go:889]: Yeah, it doesn't. No code change is required as `newCommonClusterConfigLoader` already disallows using both `--name` and `--config-file`.

**Review by cPu1 — APPROVED** [review]

**Review by aclevername — APPROVED** [review]

**Pull request merged**